opentracing/tracer.py

```diff
@@ -34,44 +33,53 @@ class Tracer(SpanPropagator):
 
     singleton_noop_span = Span()
 
-    def start_trace(self, operation_name=None, tags=None):
-        """Starts a new trace and creates a new root span.
+    def start_span(self, operation_name, parent=None, **kwargs):
```

Comment:
If we define a function with named optional parameters, their order is not fixed, e.g.

```
def start_span(self, operation_name, parent=None, tags=None, start_time=None)
```
only requires operation to be the first arg, and even that only if you're passing it as positional, passing by name can have any order.

By "compiler" I meant that if you try to call the above function with bad argument `foo=1` the call will bomb at runtime. Where as if the function used `kwargs`, the call would be valid but the function would simply ignore the argument, silently. Also, IDE can prompt for the names args, and will highlight invalid ones right away.